src/eos-updater-poll-common.c

```diff
@@ -0,0 +1,910 @@
+  return FALSE;
+}
+
+gboolean
+fetch_latest_commit (OstreeRepo *repo,
+                     GCancellable *cancellable,
+                     const gchar *remote_name,
+                     const gchar *ref,
+                     const gchar *url_override,
+                     gchar **out_checksum,
+                     EosExtensions **out_extensions,
+                     GError **error)
+{
+  g_autoptr(GBytes) summary_bytes = NULL;
+  g_autoptr(GVariant) summary = NULL;
+  g_autofree gchar *checksum = NULL;
+
+  g_return_val_if_fail (OSTREE_IS_REPO (repo), FALSE);
+  g_return_val_if_fail (cancellable == NULL || G_IS_CANCELLABLE (cancellable), FALSE);
+  g_return_val_if_fail (remote_name != NULL, FALSE);
+  g_return_val_if_fail (ref != NULL, FALSE);
+  g_return_val_if_fail (out_checksum != NULL, FALSE);
+
+  {
+    g_autoptr(GVariant) options = NULL;
+
+    options = get_repo_pull_options (url_override, ref);
+    if (!ostree_repo_pull_with_options (repo,
+                                        remote_name,
+                                        options,
+                                        NULL,
+                                        cancellable,
+                                        error))
+      return FALSE;
+  }
+
+  return fetch_commit_checksum (repo,
+                                cancellable,
+                                remote_name,
+                                ref,
+                                url_override,
+                                out_checksum,
+                                out_extensions,
+                                error);
+}
+
+static SoupURI *
+get_uri_to_sig (SoupURI *uri)
+{
+  g_autofree gchar *sig_path = NULL;
+  SoupURI *sig_uri = soup_uri_copy (uri);
+
+  sig_path = g_strconcat (soup_uri_get_path (uri), ".sig", NULL);
+  soup_uri_set_path (sig_uri, sig_path);
+
+  return sig_uri;
+}
+
+static GBytes *
+download_file (SoupURI *uri)
+{
+  g_autoptr(SoupSession) soup = NULL;
+  g_autoptr(SoupMessage) msg = NULL;
+  g_autoptr(GBytes) contents = NULL;
+  guint status = 0;
+
+  soup = soup_session_new ();
+  msg = soup_message_new_from_uri ("GET", uri);
+  status = soup_session_send_message (soup, msg);
+  if (SOUP_STATUS_IS_SUCCESSFUL (status))
+    g_object_get (msg,
+                  SOUP_MESSAGE_RESPONSE_BODY_DATA, &contents,
+                  NULL);
+
+  return g_steal_pointer (&contents);
+}
+
+gboolean
+download_file_and_signature (const gchar *url,
+                             GBytes **contents,
+                             GBytes **signature,
+                             GError **error)
+{
+  g_autoptr(SoupURI) uri = soup_uri_new (url);
+  g_autoptr(SoupURI) sig_uri = NULL;
+
+  if (uri == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "Invalid URL %s", url);
+      return FALSE;
+    }
+
+  sig_uri = get_uri_to_sig (uri);
+  *contents = download_file (uri);
+  *signature = download_file (sig_uri);
```

Comment:
Should there be some kind of error handling here if `download_file` fails? It seems like the default case here is to just put `NULL` in the outparams.